### PR TITLE
Jv-registration-validation task solution

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private final StorageDao storageDao = new StorageDaoImpl();
+
+    @BeforeEach
+    void setUp() {
+        User user1 = new User();
+        user1.setId(1L);
+        user1.setLogin("LordFar");
+        user1.setPassword("34gsge&ws");
+        user1.setAge(23);
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:27]: better create a user field to be reused across test methods
You only need to save one user in DB in the test about duplicate login - so let's save it there

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                () -> registrationService.register(actualUser3));
+
+        User actualUser4 = new User();
+        actualUser4.setLogin("Wlrensar");
+        actualUser4.setPassword("3gd4h8@44gs");
+        actualUser4.setAge(-5);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser4));
+
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:241]:
```suggestion
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        actualUser4.setLogin("Wlrensar");
+        actualUser4.setPassword("");
+        actualUser4.setAge(39);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser4));
+    }
+
+    @Test
+    void register_AllowableAge_notOk() {
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:209]:
```suggestion
    void register_underage_notOk() {
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        actualUser3.setLogin("Ignesco");
+        actualUser3.setPassword("hkl4gs&ws");
+        actualUser3.setAge(39);
+
+        assertThrows(LoginDuplicateException.class,
+                () -> registrationService.register(actualUser3));
+    }
+
+    @Test
+    void register_loginAllowableSize_notOk() {
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:139]: allowable means ok
rename everywhere

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job on your service and exceptions! See suggestions for the test class 👇

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java (hunk 2 above)
Comment (mkazarian-1) [reply src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:241]: Sorry but, I don't quite understand what this trailer means

File: src/main/java/core/basesyntax/exceptions/DataCorrectnessException.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.exceptions;
+
+public class DataCorrectnessException extends RuntimeException {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/exceptions/DataCorrectnessException.java:3]: One RegistrationException class is good enough

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 6]
```diff
@@ -2,13 +2,51 @@
...
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.DataCorrectnessException;
+import core.basesyntax.exceptions.LoginDuplicateException;
+import core.basesyntax.exceptions.NullUserDataException;
+import core.basesyntax.exceptions.NullUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int ALLOWABLE_LOGIN_SIZE = 6;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:12]:
```suggestion
    private static final int MIN_LOGIN_SIZE = 6;
```

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 7]
```diff
@@ -2,13 +2,51 @@
...
+    private static final int ALLOWABLE_PASSWORD_SIZE = 6;
+    private static final int AGE_RESTRICTION = 18;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:14]:
```suggestion
    private static final int MIN_PASSWORD_SIZE = 6;
    private static final int MIN_AGE = 18;
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+        user3.setAge(0);
+        assertThrows(DataCorrectnessException.class, () -> registrationService.register(user3));
+
+        User user4 = copeUser(correctUsersList.get(3));
+        user4.setAge(-10);
+        assertThrows(DataCorrectnessException.class, () -> registrationService.register(user4));
+
+    }
+
+    private User copeUser(User user) {
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:192]:
```suggestion
    private User copyUser(User user) {
```

**Review by Rommelua — APPROVED** [review]
